stratum/hal/lib/phal/onlp/onlpphal.cc

```diff
@@ -118,11 +118,18 @@ ::util::Status OnlpPhal::VerifyChassisConfig(const ChassisConfig& config) {
 }
 
 ::util::Status OnlpPhal::Shutdown() {
-  absl::WriterMutexLock l(&config_lock_);
-
   // TODO(unknown): add clean up code
 
-  initialized_ = false;
+  {
+    absl::WriterMutexLock l(&init_lock_);
+    if (this == singleton_) {
+      singleton_ = nullptr;
```

Comment:
This does not free the instance. On the other hand, I'm not sure if it's ever save to free the singleton, because it could still be in use somewhere. Other classes also don't free on `Shutdown`, so I'd skip it here too..